File: backend/src/Contracts/Auth/SignUpRequest.cs
namespace OptimaHealthcare.Contracts.Auth;

public sealed class SignUpRequest
{
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string MobileNumber { get; set; } = string.Empty;
    public string EmailAddress { get; set; } = string.Empty;
    public string Password { get; set; } = string.Empty;
    public int UserRoleId { get; set; }
}

File: mobile/lib/features/auth/models/signup_request.dart
class SignUpRequestModel {
  SignUpRequestModel({
    required this.firstName,
    required this.lastName,
    required this.mobileNumber,
    required this.emailAddress,
    required this.password,
    required this.userRoleId,
  });

  final String firstName;
  final String lastName;
  final String mobileNumber;
  final String emailAddress;
  final String password;
  final int userRoleId;

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'mobileNumber': mobileNumber,
      'emailAddress': emailAddress,
      'password': password,
      'userRoleId': userRoleId,
    };
  }
}
